app/lib/audit/backend.dart

```diff
@@ -26,6 +26,9 @@ AuditBackend get auditBackend => ss.lookup(#_auditBackend) as AuditBackend;
 /// Represents the backend for the audit handling and authentication.
 class AuditBackend {
   final DatastoreDB _db;
+  var _cachedRecords = _CachedRecords(DateTime(0), []);
+  Future<void>? _cacheRecordsUpdateFuture;
```

Comment:
Why not:
```suggestion
  Future<_CachedRecords> _cachedRecords = Future.value(_CachedRecords(DateTime(0), []));
```
Then you always honor the invariant by awaiting it, less risk of making a mistake.

Reply:
I've found it confusing when reading the code: we would use the same variable for (a) getting the timestamp of its update (and the value) and (b) waiting for it to be updating. I'd rather keep the "update is happening right now" variable separate, easier to write a conditional around it.